feat: Rewrite S3 SSE-C examples, add support for additional CLIs (#140)
citynetwork/docs: docs/howto/object-storage/s3/sse-c.md

* obtain [S3-compatible credentials](credentials.md).

## Creating encryption details

According to the SSE-C specification, in order to use server-side encryption, any S3 client needs to provide three pieces of information, which it includes in the request headers for each S3 request being made:

* Encryption algorithm: the only valid option here is AES256.
* Encryption key: a valid AES key, which means that key length must be 32 bytes.
* Encryption key checksum: the MD5 checksum of the encryption key.
  Your S3 API client uses this for integrity checks, and normally adds it automatically to your request.

In order to generate encryption key and store it in Barbican, proceed
as follows.

1. Generate an encryption secret:
   ```bash
   secret_raw=$(pwgen 32 1)
   ```

2. Store the secret in Barbican:
   ```bash
   barbican_secret_url=$(openstack secret store --name objectSecret --algorithm aes --bit-length 256 --payload ${secret_raw} -f value -c 'Secret href')
   ```

3. Then, whenever you need to upload or download encrypted objects, retrieve the secret from Barbican:
   ```bash
   secret=$(openstack secret get ${barbican_secret_url} -p -c Payload -f value)
   ```

> 

Author: Yup, done.

## Managing encrypted objects in S3

> Once you have your encryption secret available, you are able to create or access enabled objects.

Reviewer: ```suggestion
Once you have your encryption secret available, you can create or access enabled objects.
```

       ```bash
       mc cp \
         --recursive \
         --encrypt-key "<region>/{{brand|lower|replace(' ','')}}-encrypted=${secret}"
         ~/media/ <region>/{{brand|lower|replace(' ','')}}-encrypted
       ```
    3. Retrieve a file from S3 and decrypt it.
       Again, specify the encryption key in the same manner:
       ```bash
       mc cp \
         --encrypt-key "<region>/{{brand|lower|replace(' ','')}}-encrypted=${secret}"
         <region>/{{brand|lower|replace(' ','')}}-encrypted/file.png .
       ```

    Note that attempting to download an encrypted file *without* providing an encryption key results in an immediate HTTP 400 ("Bad Request") error:
    ```console
    $ mc cp \
      <region>/{{brand|lower|replace(' ','')}}-encrypted/file.png .
    mc: <ERROR> Unable to validate source `<region>/{{brand|lower|replace(' ','')}}-encrypted/file.png`.
    ```

=== "s3cmd"
    You cannot use `s3cmd` in combination with SSE-C.

    > `s3cmd` does contain a *client* side encryption facility, using GnuPG for encryption.
    > It also supports SSE-KMS, which is a different SSE flavor that is currently not available in {{brand}}.
=== "rclone"

>     SSE-C encryption has been implemented/fixed with version 1.54. Prior `rclone` versions won't work.

Reviewer: ```suggestion
    SSE-C encryption has been implemented/fixed with version 1.54. Earlier `rclone` versions won't work.
```
